langgraph/src/pregel/read.ts

```diff
@@ -53,55 +69,73 @@ export class ChannelRead<
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   _read(_: any, config: RunnableConfig) {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const read: (arg: string) => any = config.configurable?.[CONFIG_KEY_READ];
+    const read: (arg: string, fresh: boolean) => any =
+      config.configurable?.[CONFIG_KEY_READ];
     if (!read) {
       throw new Error(
         `Runnable ${this} is not configured with a read function. Make sure to call in the context of a Pregel process`
       );
     }
     if (Array.isArray(this.channel)) {
       const results = Object.fromEntries(
-        this.channel.map((chan) => [chan, read(chan)])
+        this.channel.map((chan) => [chan, read(chan, this.fresh)])
       );
       return results;
     }
-    return read(this.channel);
+    if (this.mapper) {
+      return this.mapper(read(this.channel, this.fresh));
+    } else {
+      return read(this.channel, this.fresh);
+    }
   }
 }
 
 const defaultRunnableBound = /* #__PURE__ */ new RunnablePassthrough();
 
-interface ChannelInvokeArgs<RunInput, RunOutput>
+interface PregelNodeArgs<RunInput, RunOutput>
   extends Partial<RunnableBindingArgs<RunInput, RunOutput>> {
   channels: Record<string, string> | string;
   triggers: Array<string>;
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  when?: (args: any) => boolean;
-  config?: RunnableConfig;
+  mapper?: (args: any) => any;
+  writers?: Runnable[];
   tags?: string[];
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  bound?: Runnable<any, any>;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  kwargs?: Record<string, any>;
+  config?: RunnableConfig;
 }
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export type ChannelInvokeInputType = any;
+export type PregelNodeInputType = any;
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export type ChannelInvokeOutputType = any;
+export type PregelNodeOutputType = any;
 
-export class ChannelInvoke<
-  RunInput = ChannelInvokeInputType,
-  RunOutput = ChannelInvokeOutputType
+export class PregelNode<
+  RunInput = PregelNodeInputType,
+  RunOutput = PregelNodeOutputType
 > extends RunnableBinding<RunInput, RunOutput, RunnableConfig> {
-  lc_graph_name = "ChannelInvoke";
+  lc_graph_name = "PregelNode";
 
   channels: Record<string, string> | string;
```

Comment:
`channels` is supposed to be type `Record<string, string> | string[]`, but I couldn't make this change without breaking a significant amount of tests. I'll make this update when I update the `Pregel` class.